zetta_utils/layer/volumetric/tools.py

```diff
@@ -117,23 +117,54 @@ def __call__(
         idx: VolumetricIndex,
         stride_start_offset_in_unit: Optional[Vec3D[int]] = None,
         mode: Literal["shrink", "expand", "exact"] = "expand",
-    ) -> Iterable[VolumetricIndex]:  # pragma: no cover # delegation, no cond
+    ) -> Iterable[VolumetricIndex]:  # pragma: no cover
+
+        bbox_strider = self._get_bbox_strider(idx, stride_start_offset_in_unit, mode)
+        if self.max_superchunk_size is not None:
+            logger.info(f"Superchunk size: {bbox_strider.chunk_size}")
+        bbox_chunks = bbox_strider.get_all_chunk_bboxes()
+        result = [
+            VolumetricIndex(
+                resolution=idx.resolution,
+                bbox=bbox_chunk,
+            )
+            for bbox_chunk in bbox_chunks
+        ]
+        return result
+
+    def get_shape(
+        self,
+        idx: VolumetricIndex,
+        stride_start_offset_in_unit: Optional[Vec3D[int]] = None,
+        mode: Literal["shrink", "expand", "exact"] = "expand",
+    ) -> Vec3D[int]:  # pragma: no cover
+
+        return self._get_bbox_strider(idx, stride_start_offset_in_unit, mode).shape
+
+    def _get_bbox_strider(
+        self,
+        idx: VolumetricIndex,
+        stride_start_offset_in_unit: Optional[Vec3D[int]] = None,
+        mode: Literal["shrink", "expand", "exact"] = "expand",
+    ) -> BBoxStrider:  # pragma: no cover
```

Comment:
This function has 3 if-else clauses, so `#pragma: no cover` shouldn't be used here. It's an internal function, so we should _not_ have a test for it, but tests for its clients (aka `self.__call__`) should cover all the branches here.